src/pkg/k8s/client/client.go

```diff
@@ -74,3 +82,65 @@ var GetCurrentContext = func() (string, string, error) {
 	}
 	return contextName, context.Cluster, nil
 }
+
+func ServeClusters(inCluster bool, currentContext string) http.HandlerFunc {
+	return func(w http.ResponseWriter, _ *http.Request) {
+		w.Header().Set("Content-Type", "text/json; charset=utf-8")
+		w.Header().Set("Cache-Control", "no-cache")
+
+		// running in cluster return []
+		if inCluster {
+			data, err := json.Marshal([]interface{}{})
+			if err != nil {
+				slog.Error("Failed to marshal empty data", "error", err)
+				http.Error(w, "JSON marshalling error", http.StatusInternalServerError)
+			}
+
+			// Write the data to the response
+			if _, err := w.Write(data); err != nil {
+				http.Error(w, "Failed to write data", http.StatusInternalServerError)
+				return
+			}
+		}
+
+		contexts, err := Contexts()
+		if err != nil {
+			slog.Error("Failed to get contexts", "error", err)
+			http.Error(w, "Failed to get contexts", http.StatusInternalServerError)
+			return
+		}
+
+		slog.Debug("ServeClusters called", "inCluster", inCluster, "currentContext", currentContext)
+
+		var clusters []ClusterInfo
+		for ctxName, context := range contexts {
+			clusters = append(clusters, ClusterInfo{Name: context.Cluster, Context: ctxName, Selected: currentContext == ctxName})
+		}
+
+		data, err := json.Marshal(clusters)
+		if err != nil {
+			slog.Error("Failed to marshal contexts", "error", err)
+			http.Error(w, "JSON marshalling error", http.StatusInternalServerError)
+			return
+		}
+
+		// Write the data to the response
+		if _, err := w.Write(data); err != nil {
+			http.Error(w, "Failed to write data", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// IsRunningInCluster checks if the application is running in cluster
+func IsRunningInCluster() (bool, error) {
+	_, err := rest.InClusterConfig()
+
+	if err == rest.ErrNotInCluster {
+		return false, nil
+	} else if err != nil {
+		return true, err
+	}
+
+	return true, nil
```

Comment:
nit: we can simplify this with 1 fewer logic check and return statement 
```suggestion
	if err == rest.ErrNotInCluster {
		return false, nil
	} 

	return true, err
```

Reply:
nice, i like that